pkg/api/api.go

```diff
@@ -183,29 +178,41 @@ func (a *API) getHeader(w http.ResponseWriter, r *http.Request) (int, error) {
 			"slot":     slot,
 			"proposer": proposer,
 		}).WithError(err).Debug("failed getHeader")
-		return http.StatusBadRequest, err
+
+		w.WriteHeader(http.StatusBadRequest)
+		_ = json.NewEncoder(w).Encode(jsonError{
+			Code:    http.StatusBadRequest,
+			Message: err.Error(),
+		})
+		return
 	}
 
 	m.Observe(a.m.RelayTiming)
 
 	if err = json.NewEncoder(w).Encode(response); err != nil {
 		a.l.WithError(err).WithField("path", r.URL.Path).Debug("failed to write response")
 		a.m.ApiReqCounter.WithLabelValues("getHeader", "500", "response encode").Inc()
-		return http.StatusInternalServerError, err
```

Comment:
Why did you delete this?

Reply:
as earlier - those functions are just plain http handlers right now.